shanalikhan/code-settings-sync
# Improve OAuth

#### Short description of what this resolves:
Notify user if OAuth listener started failed.

#### Changes proposed in this pull request:

https://github.com/shanalikhan/code-settings-sync/issues/1053#issuecomment-538248184_

**Fixes**: #1053

#### How Has This Been Tested?
1. Execute `> Sync: Reset Extension Settings`
2. Reload Visual Studio, the landing page will be shown
3. Run Thunder on Windows or use [http-server](https://www.npmjs.com/package/http-server) to listen 54321 (`hs -a :: -p 54321`)
4. Click "Login with GitHub" button, it should show a error notification at right-down corner, and won't open the authorization page
5. Click "Troubleshooting" button in the notification, it will open the wiki
6. Back to step 4 again, close the notification will do nothing (don't open the wiki)
7. Close the application that started in step 3, then retry `Login with GitHub`, it will works fine.

#### Screenshots (if appropriate):
<img width="1270" alt="Screen Shot 2020-08-10 at 8 33 20 PM" src="https://user-images.githubusercontent.com/11289158/89783065-e7b8d600-db48-11ea-998c-17a786143747.png">

#### Checklist:
